### PR TITLE
feat: check if removal key is available on backend

### PR DESCRIPTION
**Pull request by PatrykBuniX**
Before initialising mls service, check whether a removal key is available on backend.

### CODE REVIEW

File: packages/core/src/Account.ts [hunk 1]
```diff
@@ -582,7 +583,7 @@ export class Account extends TypedEventEmitter<Events> {
     });
 
     const handleMissedNotifications = async (notificationId: string) => {
-      if (this.cryptoProtocolConfig?.mls && this.backendFeatures.supportsMLS) {
+      if (this.service?.mls) {
```
Comment (PatrykBuniX) [inline packages/core/src/Account.ts:586]: It's enough to check if mls service exists - it wouldn't get initialised if mls was disabled.

File: packages/api-client/src/APIClient.ts [hunk 2]
```diff
@@ -425,4 +425,23 @@ export class APIClient extends EventEmitter {
       }
     };
   }
+
+  /**
+   * Will check if MLS is supported by backend (whether used api version supports MLS and backend removal key is present).
```
Comment (V-Gira) [inline packages/api-client/src/APIClient.ts:430]: Suggestion:
```suggestion
   * Will check if MLS is supported by backend (whether the api version used supports MLS, and whether a backend removal key is present).
```

File: packages/api-client/src/APIClient.ts [hunk 3]
```diff
@@ -425,4 +425,23 @@ export class APIClient extends EventEmitter {
...
+      return false;
+    }
+
+    let isBackendRemovalKeyPresent = false;
+    try {
+      const backendRemovalKey = (await this.api.client.getPublicKeys()).removal;
+      isBackendRemovalKeyPresent = !!backendRemovalKey;
+    } catch {}
+
+    return isBackendRemovalKeyPresent;
```
Comment (atomrc) [inline packages/api-client/src/APIClient.ts:445]: suggestion :) 

```suggestion
    try {
      const backendRemovalKey = (await this.api.client.getPublicKeys()).removal;
      return !!backendRemovalKey;
    } catch {}

    return false;
```

File: packages/api-client/src/APIClient.ts [hunk 4]
```diff
@@ -425,4 +425,23 @@ export class APIClient extends EventEmitter {
       }
     };
   }
+
+  /**
+   * Will check if MLS is supported by backend (whether used api version supports MLS and backend removal key is present).
+   */
+  public async isMLSSupportedByBackend(): Promise<boolean> {
```
Comment (atomrc) [inline packages/api-client/src/APIClient.ts:432]:
```suggestion
  public async supportsMLS(): Promise<boolean> {
```

We know we are dealing with the APIClient so no need for mentionning the backend at all in the name of the method. It's already explicit from the context

**Review by atomrc — APPROVED** [review]
LGTM. Just a few suggestions

**Pull request merged**